pytrip/cube.py

```diff
@@ -402,29 +405,106 @@ def read(self, path):
         self.read_trip_data_file(path)
 
     def read_trip_header_file(self, path):
+        """ Reads a header file, accepts also if suffix is missing, or if file"
+        is in gz compressed. User can thus specify:
+        tst001
+        tst001.hed
+        tst001.hed.gz
+
+        and in any case it will attempt to load
+        tst001.hed
+        or
+        tst001.hed.gz
+
+        Note, first the un-zipped files will be attempted to read.
+        Should these not exist, then the .gz are attempted.
+
+        Hoever, if the .hed.gz file was explicitly stated,
+        then this file will also
+        be loaded, even if a .hed is available.
+        """
+
+        is_zipped = False
         f_split = os.path.splitext(path)
+        if f_split[1] == ".gz":
+            is_zipped = True
+            f_split = os.path.splitext(f_split[0])  # get rid of .gz suffix
+        # now proper header file basename is set, but without .gz suffix.
         header_file = f_split[0] + ".hed"
+
+        # Does the .hed file exist? If not retry with .hed.gz suffix
+        # PyTRiP will also attempt to load .hed.gz files if user did
+        # not specify the .gz suffix explicitly
+        if os.path.isfile(header_file) is False:
```

Comment:
` if os.path.isfile(header_file) is False:` can be replaced with  `if not os.path.isfile(header_file)`: